### PR TITLE
Add TCP Flow test

### PR DESCRIPTION
**Pull request by gargnupur**
Add support for TCP server and a tcp flow test
Also added support to add new filters in http filters

### CODE REVIEW

File: src/envoy/http/authn/filter_context.cc [hunk 1]
```diff
@@ -102,15 +102,17 @@ bool FilterContext::getJwtPayloadFromEnvoyJwtFilter(const std::string& issuer,
     return false;
   }
 
-  // Serialize the payload from Envoy jwt filter first before writing it to |payload|.
-  // TODO (pitlv2109): Return protobuf Struct instead of string, once Istio jwt filter is removed.
-  // Also need to change how Istio authn filter processes the jwt payload.
+  // Serialize the payload from Envoy jwt filter first before writing it to
```
Comment (mandarjog) [inline src/envoy/http/authn/filter_context.cc]: nit: why are all these changes showing up here? are we using a different formatting ?

Comment (gargnupur) [reply src/envoy/http/authn/filter_context.cc]: I ran scripts/run_gofmt.sh that we are using in istio.io/istio... looks like it was added only 7 days back.. I can revert these changes if we don't intend to use this script for formatting?

Comment (PiotrSikora) [reply src/envoy/http/authn/filter_context.cc]: `make check` is broken (see #2287), so the last few commits were committed without formatting being properly enforced.

Comment (PiotrSikora) [reply src/envoy/http/authn/filter_context.cc]: Fixed, please merge with `master`.

Comment (gargnupur) [reply src/envoy/http/authn/filter_context.cc]: Thanks @PiotrSikora

File: test/envoye2e/env/tcp_envoy_conf.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  address:
+    socket_address:
+      address: 127.0.0.1
+      port_value: {{.Ports.ClientAdminPort}}
+static_resources:
+  clusters:
+  - name: client
+    connect_timeout: 5s
+    type: STATIC
+    hosts:
```
Comment (kyessenov) [inline test/envoye2e/env/tcp_envoy_conf.go]: `hosts` is deprecated, please update the config file. I think the new way is similar to EDS load assignment API.

**Comment by kyessenov** [comment]
I can LGTM once the static bootstrap is updated to change the deprecated field.https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/cds.proto#L191

File: test/envoye2e/env/tcp_envoy_conf.go (hunk 2 above)
Comment (gargnupur) [reply test/envoye2e/env/tcp_envoy_conf.go]: Fixed.. also fixed it in envoy_conf.go

**Pull request merged**